### PR TITLE
ARM64-SVE: Add `GatherVectorWithByteOffsetFirstFaulting`

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Part of #99957. Adds a new test template for the `ByteOffset` first-faulting APIs. All stress tests are passing.

@dotnet/arm64-contrib PTAL, thanks!

### CODE REVIEW

File: src/coreclr/jit/hwintrinsiclistarm64sve.h [hunk 1]
```diff
@@ -121,6 +121,7 @@ HARDWARE_INTRINSIC(Sve,           GatherVectorUInt16WithByteOffsetsZeroExtend,
...
 HARDWARE_INTRINSIC(Sve,           GatherVectorUInt32ZeroExtend,                                     -1,     -1,     {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_ld1w,       INS_sve_ld1w,       INS_sve_ld1w,       INS_sve_ld1w,       INS_invalid,        INS_invalid},     HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_SpecialCodeGen|HW_Flag_ExplicitMaskedOperation|HW_Flag_LowMaskedOperation)
+HARDWARE_INTRINSIC(Sve,           GatherVectorWithByteOffsetFirstFaulting,                          -1,     -1,     {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_ldff1w,     INS_sve_ldff1w,     INS_sve_ldff1d,     INS_sve_ldff1d,     INS_sve_ldff1w,     INS_sve_ldff1d},  HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_SpecialCodeGen|HW_Flag_ExplicitMaskedOperation|HW_Flag_LowMaskedOperation|HW_Flag_SpecialSideEffectMask)
```
Comment (jakobbotsch) [inline src/coreclr/jit/hwintrinsiclistarm64sve.h:124]:
```suggestion
HARDWARE_INTRINSIC(Sve,           GatherVectorWithByteOffsetFirstFaulting,                          -1,     -1,     {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_ldff1w,     INS_sve_ldff1w,     INS_sve_ldff1d,     INS_sve_ldff1d,     INS_sve_ldff1w,     INS_sve_ldff1d},  HW_Category_MemoryLoad,            HW_Flag_Scalable|HW_Flag_SpecialCodeGen|HW_Flag_ExplicitMaskedOperation|HW_Flag_LowMaskedOperation|HW_Flag_SpecialSideEffect_Other)
```
(I am fixing the others in #105973.)

**Review by a74nh — APPROVED** [review]
This LGTM

**Comment by amanasifkhalid** [comment]
Thanks for the review! @TIHan could you PTAL? Thanks

**Review by AndyAyersMS — APPROVED** [review]
Changes look good to me, though I am not familiar with much of this.

File: src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorByteOffsetFirstFaulting.template [hunk 2]
```diff
@@ -0,0 +1,764 @@
...
+                // when falseReg == targetReg because this instruction uses Pg/Z to update the targetReg
+                // instead of Pg/M to merge it. As such, the value of falseReg is lost. Ideally, such
+                // instructions should be marked similar to RMW (a different flag name) to make sure that
+                // we do not assign falseReg/targetReg same. Then, we would do something like this:
+                //
+                // ldnf1sh target, pg/z, [x0]
+                // sel mask, target, target, falseReg
+                //
+                // This needs more careful thinking, so disabling it for now.
+                // test.ConditionalSelect_FalseOp();
```
Comment (kunalspathak) [inline src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorByteOffsetFirstFaulting.template:73]: Will you send a follow-up PR to uncomment this once [https://github.com/dotnet/runtime/pull/105737](https://github.com/dotnet/runtime/pull/105737) is merged or can you merge [https://github.com/dotnet/runtime/pull/105737](https://github.com/dotnet/runtime/pull/105737) first and uncomment this one?

**Review by kunalspathak — APPROVED** [review]
LGTM modulo the comment about one of the test case.

File: src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorByteOffsetFirstFaulting.template (hunk 2 above)
Comment (amanasifkhalid) [reply src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorByteOffsetFirstFaulting.template:73]: This scenario is also disabled in [`SveGatherVectorFirstFaultingIndices.template`](https://github.com/dotnet/runtime/blob/adb1fee95cac68b1a64561c9f414966964fee408/src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorFirstFaultingIndices.template#L73) and [`SveGatherVectorFirstFaultingVectorBases.template`](https://github.com/dotnet/runtime/blob/adb1fee95cac68b1a64561c9f414966964fee408/src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorFirstFaultingVectorBases.template#L73), so maybe we can re-enable it across all templates as a follow-up to #105737?

Comment (amanasifkhalid) [reply src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorByteOffsetFirstFaulting.template:73]: Created an issue for this: #106279

**Pull request merged**